src/scrabble.js

```diff
@@ -1,5 +1,130 @@
-function scrabble() {
-  // write code here
+const { LETTER_SCORES, INCORRECT_TOKENS } = require('./constants')
+let {
+  MULTIPLE,
+  VALID_SCORE,
+  MULTIPLIER_USAGE,
+  SCORE_MULTIPLIER
+} = require('./constants')
+
+const stringToArray = (word) => {
+  if (typeof word === 'string') {
+    return word.toUpperCase().split('')
+  } else {
+    return ['']
+  }
+}
+
+const stringArrayToScore = (stringArray) => {
+  let numArray = stringArray.map((val) => {
+    if (multiplyCheck(val)) {
+      MULTIPLE = scoreMultiplicatorizer(val)
+      return undefined
+    } else {
+      return findLetterScore(val)
+    }
+  })
+  numArray = numArray.filter((val) => !(val === undefined))
+  return numArray
+}
+
+const multiplyCheck = (character) => {
+  edgeCasesIDisagreeWith(character)
+  logMultiplierUsage(character)
+  if (SCORE_MULTIPLIER[character]) {
+    return true
+  } else {
+    return false
+  }
+}
+
+const findLetterScore = (character) => {
+  const result = LETTER_SCORES[character] * MULTIPLE
+  if (result) {
+    return result
+  } else {
+    return 0
+  }
+}
+
+const scoreMultiplicatorizer = (character) => {
+  const multiplier = SCORE_MULTIPLIER[character]
+  if (multiplier) {
+    return MULTIPLE * multiplier
+  } else {
+    return MULTIPLE
+  }
+}
+
+const logMultiplierUsage = (character) => {
+  const usageKeys = Object.keys(MULTIPLIER_USAGE)
+  if (usageKeys.includes(character)) {
+    MULTIPLIER_USAGE[character]++
+  }
+}
+
+const edgeCasesIDisagreeWith = (character) => {
+  if (INCORRECT_TOKENS.includes(character)) {
+    VALID_SCORE = false
+  }
+}
+
+const tallyScore = (numArray) => {
+  if (numArray.length === 0) {
+    return 0
+  } else {
+    return numArray.reduce((acc, val) => acc + val)
```

Comment:
see, I knew I'd learn something from this – I'm gonna look into what this does

Reply:
First thing I do when dealing with a data type is just remind myself what methods it has. `.reduce` was perfect for finding sums in this case.